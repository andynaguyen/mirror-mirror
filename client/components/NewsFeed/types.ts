import { QueryProps } from 'react-apollo';
import { NewsFeedData } from '../../../common/types';

interface NewsFeedProps extends ApolloWrapperProps {
  data?: QueryProps<NewsFeedData>;
  newsFeed: NewsFeedData;
}

interface ApolloWrapperProps {
  visible: boolean;
}

type NewsFeedResponse = {
  newsFeed: NewsFeedData;
};

export { NewsFeedProps, ApolloWrapperProps, NewsFeedResponse };
