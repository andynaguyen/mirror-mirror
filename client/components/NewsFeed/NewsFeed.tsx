import * as React from 'react';
import styled from 'styled-components';
import { NewsFeedProps } from './types';

const FeedList = styled.ul`
  padding: 0;
`;

const Headline = styled.li`
  font-size: 14px;
`;

export default ({ newsFeed, visible }: NewsFeedProps) => {
  const { feed } = newsFeed;
  return visible ? <FeedList>{feed.map(({ title }) => <Headline>{title}</Headline>)}</FeedList> : null;
};
