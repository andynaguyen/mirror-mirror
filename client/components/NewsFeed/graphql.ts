import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { NewsFeedResponse, ApolloWrapperProps } from './types';

const query = gql`
  query {
    newsFeed {
      feed {
        title
        description
      }
    }
  }
`;

export default graphql<NewsFeedResponse, ApolloWrapperProps>(query, {
  options: {
    pollInterval: 1000 * 60 * 15, // 15 min
    fetchPolicy: 'network-only',
  },
  props: ({ data: { newsFeed, error } }) => ({
    newsFeed: {
      feed: error || !newsFeed ? [] : newsFeed.feed,
    },
  }),
});
