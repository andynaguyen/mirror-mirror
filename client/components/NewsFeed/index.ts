import NewsFeed from './NewsFeed';
import withGraphql from './graphql';

/**
 * A component showing trending news headlines updated every 15 min.
 */
const EnhancedNewsFeed = withGraphql(NewsFeed);
export default EnhancedNewsFeed;
