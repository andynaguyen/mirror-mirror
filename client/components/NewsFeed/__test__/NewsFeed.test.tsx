import 'jest-styled-components';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import NewsFeed from '../NewsFeed';

describe('<NewsFeed />', () => {
  const newsFeedData = {
    feed: [
      {
        title: 'duck',
        description: 'duck',
      },
      {
        title: 'duck',
        description: 'duck',
      },
      {
        title: 'goose',
        description: 'goose',
      },
    ],
  };

  it('should render the contents of the news feed component', () => {
    const component = renderer.create(<NewsFeed newsFeed={newsFeedData} visible />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render nothing', () => {
    const component = renderer.create(<NewsFeed newsFeed={newsFeedData} visible={false} />);
    const tree = component.toJSON();
    expect(tree).toBeNull();
  });
});
