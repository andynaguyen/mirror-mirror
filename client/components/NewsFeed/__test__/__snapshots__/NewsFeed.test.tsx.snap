// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsFeed /> should render the contents of the news feed component 1`] = `
.c0 {
  padding: 0;
}

.c1 {
  font-size: 14px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    duck
  </li>
  <li
    className="c1"
  >
    duck
  </li>
  <li
    className="c1"
  >
    goose
  </li>
</ul>
`;
